src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterHelper.cs

```diff
@@ -0,0 +1,228 @@
+{
+    public partial class BinaryFormatterTests
+    {
+        private static void CheckForAnyEquals(object obj, object deserializedObj)
+        {
+            if (obj != null && deserializedObj != null)
+            {
+                object equalityResult = null;
+                Type objType = obj.GetType();
+
+                // Check if custom equality extension method is available
+                MethodInfo customEqualityCheck = GetExtensionMethod(typeof(EqualityExtensions).Assembly, objType);
+                if (customEqualityCheck != null)
+                {
+                    equalityResult = customEqualityCheck.Invoke(obj, new object[] { obj, deserializedObj });
+                }
+                else
+                {
+                    // Check if object.Equals(object) is overridden and if not check if there is a more concrete equality check implementation
+                    bool equalsNotOverridden = objType.GetMethod("Equals", new Type[] { typeof(object) }).DeclaringType == typeof(object);
+                    if (equalsNotOverridden)
+                    {
+                        // If type doesn't override Equals(object) method then check if there is a more concrete implementation
+                        // e.g. if type implements IEquatable<T>.
+                        MethodInfo equalsMethod = objType.GetMethod("Equals", new Type[] { objType });
+                        if (equalsMethod.DeclaringType != typeof(object))
+                        {
+                            equalityResult = equalsMethod.Invoke(obj, new object[] { deserializedObj });
+                        }
+                    }
+                }
+
+                if (equalityResult != null)
+                {
+                    Assert.True((bool)equalityResult, "Error during equality check of type " + objType.FullName);
+                    return;
+                }
+            }
+
+            try
+            {
+                Assert.Equal(obj, deserializedObj);
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("Error during equality check of type " + obj?.GetType()?.FullName);
+                throw;
+            }
+        }
+
+        private static MethodInfo GetExtensionMethod(Assembly assembly, Type extendedType)
+        {
+            if (extendedType.IsGenericType)
+            {
+                return typeof(EqualityExtensions).GetMethods()
+                    ?.SingleOrDefault(m => m.Name == "IsEqual" && m.GetParameters().Length == 2 && m.GetParameters()[0].ParameterType.Name == extendedType.Name)
+                    ?.MakeGenericMethod(extendedType.GenericTypeArguments[0]);
+            }
+            else
+            {
+                return typeof(EqualityExtensions).GetMethod("IsEqual", new[] { extendedType, extendedType });
+            }
+        }
+
+        public static string GetTestDataFilePath()
+        {
+            string GetRepoRootPath()
```

Comment:
No, it's just C#7 sugar... The method isn't needed anywhere else so it makes sense to inline it.